### PR TITLE
intermediate_source/torch_compile_conv_bn_fuser.py 번역

### PR DESCRIPTION
**Pull request by skt0725**
## 라이선스 동의
_변경해주시는 내용에 BSD 3항 라이선스가 적용됨을 동의해주셔야 합니다._<br />
_더 자세한 내용은 [기여하기 문서](https://github.com/PyTorchKorea/tutorials-kr/blob/master/CONTRIBUTING.md)를 참고해주세요._<br />
_동의하시면 아래 `[ ]`를 `[x]`로 만들어주세요._<br />

- [x] 기여하기 문서를 확인하였으며, 본 PR 내용에 BSD 3항 라이선스가 적용됨에 동의합니다.

## 관련 이슈 번호
_이 Pull Request와 관련있는 이슈 번호를 적어주세요._<br />
_이슈 또는 PR 번호 앞에 #을 붙이시면 제목을 바로 확인하실 수 있습니다. (예. #999 )_

- **이슈 번호**: #1109 

## PR 종류
_이 PR에 해당되는 종류 앞의 `[ ]`을 `[x]`로 변경해주세요._<br />
- [ ] 오탈자를 수정하거나 번역을 개선하는 기여
- [x] 번역되지 않은 튜토리얼을 번역하는 기여
- [ ] 공식 튜토리얼 내용을 반영하는 기여
- [ ] 위 종류에 포함되지 않는 기여

## PR 설명
_이 PR로 무엇이 달라지는지 대략적으로 알려주세요._

intermediate_source/torch_compile_conv_bn_fuser.py 문서를 새로 번역하였습니다.

### CODE REVIEW

**Comment by skt0725** [comment]
원문 내용 중 같은 문단 제목이 반복하여 등장하는 것을 확인하였습니다. (Fusing Convolution with Batch Norm)
현재 버전에서는 원문 그대로 해당 문단 제목을 두 번 넣어두었으나,
실수일 가능성이 있어 이 부분에 대한 논의가 필요할 것 같습니다!

**Comment by hijyun** [comment]
자연스럽게 번역을 잘 진행하신 것 같습니다.
LGTM !

**Comment by NamsanMan** [comment]
번역하시느라 고생 많으셨습니다.
전체적으로 기술적인 흐름이 잘 유지되어 있고, 코드 설명도 이해하기 쉽게 번역된 것 같습니다!

아래는 이미 다른 분께서 남겨주신 문단 제목 중복 관련 내용과 겹치지 않는 부분 위주로 추가 제안드립니다.

line 152의 "패턴 매처는 형태에 구애받지 않습니다!"는 의미는 전달되지만, 원문의 shape-agnostic 의미를 살리려면 "패턴 매처는 입력 형태에 독립적입니다!" 또는 "패턴 매처는 입력 형태에 구애받지 않습니다!"로 수정하는 것이 더 명확해 보입니다.

line 157의 "모두 출력 채널과 일치하는 형태(num_features,)를 가짐"은 문장 종결이 다소 어색하므로 "모두 출력 채널과 일치하는 형태(num_features,)를 가집니다."로 수정하면 더 자연스러울 것 같습니다.

line 282의 "무궁무진한 일들 중 하나의 예시일 뿐입니다"는 원문보다 의미가 다소 확장된 표현처럼 느껴집니다. "가능한 활용 방식 중 하나의 예시일 뿐입니다" 정도로 번역하면 더 담백하고 원문 의미에 가까울 것 같습니다.

**Comment by jykimai** [comment]
번역하시느라 고생 많으셨습니다! 
코드 주석까지 꼼꼼히 번역해주신 점이 인상 깊었습니다.

전반적으로 자연스럽게 잘 번역해주셨고, 두 부분에 대해 다듬어보면 어떨까 하여 제안드립니다:)


**라인**: line 267 (퓨전 패스 테스트하기 섹션 내부)

- **현재 번역**:
  "# 우리의 커스텀 패턴을 적용하여 컴파일합니다."

- **근거**:
  원문 `our custom pattern`의 `our`가 그대로 옮겨져 '우리의'라는 소유격이 들어가 있는데, 한국어 기술 문서에서는 인칭대명사를 생략하거나 다른 표현으로 풀어주는 것이 좀 더 자연스럽게 느껴질 수 있습니다.

- **제안**:
  "# 커스텀 패턴을 적용하여 컴파일합니다."

  또는 앞에서 패턴을 정의했음을 명시하고 싶다면 "# 앞서 정의한 커스텀 패턴을 적용하여 컴파일합니다." 도 좋아 보입니다.


**라인**: line 244 (ResNet18 성능 측정 섹션)

- **현재 번역**:
  "# ResNet18과 같은 더 큰 모델에 우리의 퓨전 최적화 단계를 테스트하여"

- **근거**:
  같은 `fusion pass`라는 개념이 앞 섹션 제목(line 196 부근 '퓨전 패스 테스트하기')에서는 '퓨전 패스'로 번역되었는데, 이 라인에서는 '퓨전 최적화 단계'로 표현되어 있어 용어가 다르게 쓰인 것처럼 보일 수 있어서 통일이 필요해 보입니다.
추가로 '우리의'도 위 제안과 같은 맥락에서 자연스럽게 표현하면 좋을 것 같습니다.

- **제안**:
  "# ResNet18과 같은 더 큰 모델에 이 퓨전 패스를 테스트하여"


다시 한번 번역 수고 많으셨습니다!

**Comment by skt0725** [comment]
@NamsanMan @hijyun @jykimai 
지금까지 코멘트 반영하여 수정하였습니다.
시간내주셔서 피드백 주셔서 감사합니다!

File: intermediate_source/torch_compile_conv_bn_fuser.py [hunk 1]
```diff
@@ -1,32 +1,31 @@
 # -*- coding: utf-8 -*-
 """
-Building a Convolution/Batch Norm fuser with torch.compile
+torch.compile 기반 합성곱·배치 정규화 퓨저(Convolution/Batch Norm fuser) 만들기
 ===========================================================
```
Comment (hyoyoung) [inline intermediate_source/torch_compile_conv_bn_fuser.py]: 문서 제목뒤에는 ===가 제목 글보다 더 길거나 같아야합니다

File: intermediate_source/torch_compile_conv_bn_fuser.py [hunk 2]
```diff
@@ -1,32 +1,31 @@
...
 
-**Author:** `Horace He <https://github.com/chillee>`_, `Will Feng <https://github.com/yf225>`_
+**저자:** `Horace He <https://github.com/chillee>`_, `Will Feng <https://github.com/yf225>`_ 번역: `심기택 <https://github.com/skt0725>`_
```
Comment (hyoyoung) [inline intermediate_source/torch_compile_conv_bn_fuser.py]: 번역은 한칸 아래로 내려서 추가 부탁드립니다.

File: intermediate_source/torch_compile_conv_bn_fuser.py [hunk 3]
```diff
@@ -270,23 +261,22 @@ def benchmark(model, iters=20):
...
 compiled_with_pattern_matching = torch.compile(rn18, backend="inductor")
 
-# Benchmark compiled model
+# 컴파일된 모델의 성능을 측정합니다.
 print("\ntorch.compile (with conv-bn pattern matching and other fusions): ", benchmark(compiled_with_pattern_matching))
 
 
-############
-# Conclusion
+######################################################################
```
Comment (hyoyoung) [inline intermediate_source/torch_compile_conv_bn_fuser.py]: 불필요하게 많은 ###을 넣을 필요는 없습니다

File: intermediate_source/torch_compile_conv_bn_fuser.py [hunk 4]
```diff
@@ -187,48 +179,47 @@ def conv_bn_replacement(x, conv_weight, conv_bias, bn_mean, bn_var, bn_weight, b
...
 
-# Ensure pattern matcher is enabled
+# 패턴 매처가 활성화되어 있는지 확인합니다.
 config.pattern_matcher = True
 
 fused_model = torch.compile(model, backend="inductor")
 inp = torch.randn(5, 1, 1, 1).to(device)
 
-# Run the model to trigger compilation and pattern matching
+# 모델을 실행하여 컴파일과 패턴 매칭 과정을 동작시킵니다.
```
Comment (hyoyoung) [inline intermediate_source/torch_compile_conv_bn_fuser.py]: 동작시킵니다는 약간 어색해보입니다
과정이 수행되도록 합니다. 정도는 어떨까요

**Review by hyoyoung — CHANGES_REQUESTED** [review]
전반적으로 잘되어있으나 사소한 문서 포맷 수정 사항이 있습니다
확인하고 수정 부탁드립니다

File: intermediate_source/torch_compile_conv_bn_fuser.py (hunk 4 above)
Comment (skt0725) [reply intermediate_source/torch_compile_conv_bn_fuser.py]: 모델을 실행하여 컴파일과 패턴 매칭 과정을 수행합니다
이렇게 수정하였습니다!

File: intermediate_source/torch_compile_conv_bn_fuser.py (hunk 3 above)
Comment (skt0725) [reply intermediate_source/torch_compile_conv_bn_fuser.py]: 수정하였습니다!

File: intermediate_source/torch_compile_conv_bn_fuser.py (hunk 1 above)
Comment (skt0725) [reply intermediate_source/torch_compile_conv_bn_fuser.py]: 제목 글보다 더 길게 추가하였습니다!

File: intermediate_source/torch_compile_conv_bn_fuser.py (hunk 2 above)
Comment (skt0725) [reply intermediate_source/torch_compile_conv_bn_fuser.py]: 반영하여 수정하였습니다!

**Comment by skt0725** [comment]
> 전반적으로 잘되어있으나 사소한 문서 포맷 수정 사항이 있습니다 확인하고 수정 부탁드립니다

피드백 감사합니다!
수정 반영하여 확인부탁드립니다.

**Review by hyoyoung — APPROVED** [review]
LGTM

File: intermediate_source/torch_compile_conv_bn_fuser.py [hunk 5]
```diff
@@ -36,10 +36,10 @@
 device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
 
 ######################################################################
-# For this tutorial, we are going to create a model consisting of convolutions
-# and batch norms. Note that this model has some tricky components - some of
-# the conv/batch norm patterns are hidden within Sequentials and one of the
-# ``BatchNorms`` is wrapped in another Module.
+# 이번 튜토리얼에서는 합성곱과 배치 정규화로 구성된 모델을 만들어 보겠습니다.
+# 이 모델에는 몇 가지 까다로운 요소가 있다는 점에 유의하세요.
+# 일부 합성곱·배치 정규화 패턴은 Sequential 내부에 숨겨져 있으며, 배치 정규화 중 하나는 또 다른
```
Comment (jih0-kim) [inline intermediate_source/torch_compile_conv_bn_fuser.py]: '배치 정규화'로 번역하신 ``BatchNorms`` 는 특정 모듈/클래스 명칭을 나타내는 코드 스타일로 표현되었기 때문에, 원문과 같이 코드 스타일로 표현하는 게 더 적절해 보입니다.

File: intermediate_source/torch_compile_conv_bn_fuser.py [hunk 6]
```diff
@@ -131,14 +126,13 @@ def fuse_conv_bn_weights(conv_w, conv_b, bn_rm, bn_rv, bn_eps, bn_w, bn_b):
...
+# torch.compile 기반 패턴 매칭
 # ------------------------------------
-# Now that we have our fusion logic, we need to register a pattern that
-# torch.compile's pattern matcher will recognize and replace during
-# compilation.
+# 이제 퓨전 로직을 구현했으므로 컴파일 과정에서 torch.compile의 패턴 매처가 인식하고 치환할 수 있는
+# 패턴을 등록해야 합니다.
 
-# Define the pattern we want to match: conv2d followed by batch_norm
+# 매칭하려는 패턴을 정의합니다: conv2d 다음에 batch_norm이 오는 패턴입니다.
```
Comment (jih0-kim) [inline intermediate_source/torch_compile_conv_bn_fuser.py]: 콜론(:)은 마침표(.)로 표기하는 것이 더 자연스러워 보입니다.

**Review by jih0-kim — CHANGES_REQUESTED** [review]
번역 문서를 꼼꼼히 완성해 주셔서 감사합니다. 
몇 가지 요청 사항 확인 부탁드립니다. 
용어집을 참고하시면 tensor의 경우 별도 번역 없이 그대로 표기하는 것으로 약속하고 있기 때문에 이 부분도 일괄 검토해 주세요.

**Comment by skt0725** [comment]
> 번역 문서를 꼼꼼히 완성해 주셔서 감사합니다. 몇 가지 요청 사항 확인 부탁드립니다. 용어집을 참고하시면 tensor의 경우 별도 번역 없이 그대로 표기하는 것으로 약속하고 있기 때문에 이 부분도 일괄 검토해 주세요.

@jih0-kim 시간내어 피드백 주셔서 감사드립니다.
tensor 표기와 요청 사항 확인하여 수정하였습니다.

File: intermediate_source/torch_compile_conv_bn_fuser.py [hunk 7]
```diff
@@ -36,10 +36,10 @@
 device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
 
 ######################################################################
-# For this tutorial, we are going to create a model consisting of convolutions
-# and batch norms. Note that this model has some tricky components - some of
-# the conv/batch norm patterns are hidden within Sequentials and one of the
-# ``BatchNorms`` is wrapped in another Module.
+# 이번 튜토리얼에서는 합성곱과 배치 정규화로 구성된 모델을 만들어 보겠습니다.
+# 이 모델에는 몇 가지 까다로운 요소가 있다는 점에 유의하세요.
+# 일부 합성곱·배치 정규화 패턴은 Sequential 내부에 숨겨져 있으며, BatchNorms 중 하나는 또 다른
```
Comment (jih0-kim) [inline intermediate_source/torch_compile_conv_bn_fuser.py]: BatchNorms 로 작성하는 것과 ``BatchNorms`` 로 작성하는 것은 다릅니다.
원문에서와 같이 "코드 스타일"의 표기가 그대로 유지될 수 있도록 수정 요청드려요!

**Review by jih0-kim — CHANGES_REQUESTED** [review]
코멘트 내용 확인 부탁드립니다~! 수정 요청드린 사항 반영 감사해요!

File: intermediate_source/torch_compile_conv_bn_fuser.py (hunk 7 above)
Comment (skt0725) [reply intermediate_source/torch_compile_conv_bn_fuser.py]: @jih0-kim 코드 스타일 표기로 수정하여 반영했습니다! 감사합니다!

**Review by jih0-kim — APPROVED** [review]
LGTM
